actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -203,25 +225,65 @@ module ClassMethods
       #     class ApplicationController < ActionController::Base
       #       protect_from_forgery store: CustomStore.new
       #     end
+      #
+      # *   `:using` - Set the verification strategy for CSRF protection.
+      #
+      #     Built-in verification strategies are:
+      #
+      #     *   `:header_only` - Uses the `Sec-Fetch-Site` header sent by modern
+      #         browsers to verify that requests originate from the same site. This
+      #         approach does not require authenticity tokens but only works with
+      #         browsers that support the Fetch Metadata Request Headers. Requests
+      #         without a valid `Sec-Fetch-Site` header will be rejected. This is
+      #         the default.
+      #
+      #     *   `:header_or_legacy_token` - A hybrid approach that first checks the
+      #         `Sec-Fetch-Site` header. If the header indicates same-site or
+      #         same-origin, the request is allowed. Requests with a cross-site
+      #         value are rejected. When the header is missing or "none", it falls
+      #         back to checking the authenticity token. This mode logs when
+      #         falling back to help identify requests that should be fixed to work
+      #         with `:header_only`. Use this if you need to support older browsers
+      #         that don't send the `Sec-Fetch-Site` header.
+      #
+      # *   `:trusted_origins` - Array of origins to allow for cross-site requests,
+      #     such as OAuth/SSO callbacks, third-party embeds, and legitimate remote
+      #     form submission.
+      #
+      # Example:
+      #
+      #     class ApplicationController < ActionController::Base
+      #       # Modern browsers only (default)
+      #       protect_from_forgery using: :header_only, with: :exception
+      #
+      #       # Hybrid approach with fallback for older browsers
+      #       protect_from_forgery using: :header_or_legacy_token, with: :exception
+      #
+      #       # Allow cross-site requests from trusted origins
+      #       protect_from_forgery trusted_origins: %w[ https://accounts.google.com ]
+      #     end
       def protect_from_forgery(options = {})
         options = options.reverse_merge(prepend: false)
 
         self.forgery_protection_strategy = protection_method_class(options[:with] || :null_session)
         self.request_forgery_protection_token ||= :authenticity_token
 
         self.csrf_token_storage_strategy = storage_strategy(options[:store] || SessionStore.new)
+        self.forgery_protection_verification_strategy = verification_strategy(options[:using] || :header_only)
```

Comment:
I believe this should be 
```suggestion
        self.forgery_protection_verification_strategy = verification_strategy(options[:using] || forgery_protection_verification_strategy)
```

For those who still want to keep the old behaviour